workspace/solution.go

```diff
@@ -67,17 +69,15 @@ func (s *Solution) Write(dir string) error {
 	if err != nil {
 		return err
 	}
-
-	path := filepath.Join(dir, solutionFilename)
-
-	// Hack because ioutil.WriteFile fails on hidden files
-	visibility.ShowFile(path)
-
-	if err := ioutil.WriteFile(path, b, os.FileMode(0600)); err != nil {
+	metadataAbsoluteFilepath := NewExerciseFromDir(dir).MetadataFilepath()
```

Comment:
'metadataAbsoluteFilepath' is a bit awkward but 'metadataFilepath' would shadow the var in Solution.